keps/sig-apps/4017-pod-index-label/README.md

```diff
@@ -0,0 +1,898 @@
+where it adds the corresponding annotation.
+
+Next, we must determine how enabling this new feature will interact with existing workloads. There are a couple of options:
+
+1. Only inject the label on *newly created pods*, so an existing StatefulSet/Indexed Jobs may include pods with the label and some without it. 
+This means for the user to utilize the label via the downard API, or to use the label for pod selection, they will need to recreate 
+the StatefulSet so the label is present on all pods.
+
+2. Inject the label only on pods for *newly created StatefulSets/Indexed Jobs*. We can track this by annotating StatefulSets/Indexed Jobs. This way,
+for a given StatefulSet/IndexedJob, either none of the pods have this label, or all of them do, which will provide a more consistent
+user experience. However, in the case of a cluster downgrade to a version without this feature, new pods would start getting
+created without this label again.
+
+Both options will not be disruptive to existing workloads, however, I lean towards option 2 so outside of the cluster downgrade edge case,
+there will be consistency within a single StatefulSet/Indexed Job and therefore a more predictable user experience.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[ ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
```

Comment:
I would fill this out.  It seems like you have a good idea of what code you would modify so you could fill this out.